addons/survey/controllers/main.py

```diff
@@ -80,6 +80,14 @@ def _check_validity(self, survey_token, answer_token, ensure_token=True):
         if (not survey_sudo.page_ids and survey_sudo.questions_layout == 'page_per_section') or not survey_sudo.question_ids:
             return 'survey_void'
 
+        if answer_sudo:
+            if request.env.user._is_public() and answer_sudo.partner_id:
+                # answers from public user should not have any partner_id; this indicates probably a cookie issue
+                return 'answer_wrong_user'
+            if not request.env.user._is_public() and answer_sudo.partner_id != request.env.user.partner_id:
+                # partner mismatch, probably a cookie issue
```

Comment:
Hi @oco-odoo,

I found a possible collateral issue with this fix.

Since now it's validated current user matches the answer's user, it's not possible anymore for an internal user to print an answer of another user, even if the user trying to print is a survey manager.

Is this expected?

I'm guessing no, because if that wasn't possible, at least a warning message should be displayed, instead of assuming a cookie issue.

Regards,

CC @hugho-ad @efrainmacias

Reply:
Hi @luisg123v ,
This has been reported by other people on the commit already, so we're aware :) @dbeguin is on it.

Reply:
@luisg123v https://github.com/odoo/odoo/pull/85583